src/data/markdown/docs/30 xk6-browser/02 xk6-browser.md

```diff
@@ -0,0 +1,124 @@
+
+xk6-browser aims to provide rough compatibility with the Playwright API, so you don’t need to learn a completely new API.
+
+## Why xk6-browser?
+
+The [performance golden rule](https://www.stevesouders.com/blog/2012/02/10/the-performance-golden-rule/) states that 80%-90% of the website response time is spent on the frontend. But when the number of concurrent users is high, the percentage of time spent between the frontend and backend changes:
+
+![Frontend time vs Backend time with increasing amount of website visitors](./images/Frontend-Backend-LoadTesting.png)
+
+While backend performance testing is useful and efficient, it doesn’t represent the overall user experience.
+
+With xk6-browser, you can interact with the browser to script tests much closer to the actual website experience. You'll get insight into what your user sees on a browser-level and catch issues skipped from the protocol level.
+
+At the same time, you can adopt a hybrid approach to performance testing by leveraging the existing functionalities of k6 and perform backend load testing on a protocol-level, all in the same script. This gives you a full picture of how the user experience is under different load scenarios.
+
+## Use case for xk6-browser
+
+The main use case for xk6-browser is to test performance on the browser level. Browser-level testing  provides a way to measure user experience and  find issues that are difficult to catch on the protocol level. Browser-level testing can help you answer questions like:
+
+- When my application is receiving thousands of simultaneous requests from the protocol-level, what happens to the frontend?
+- How can I get metrics specific to browsers, like total page load time?
+- Are all my elements interactive on the frontend?
+- Are there any loading spinners that take a long time to disappear?
+
+## A simple test using xk6-browser
+
+<CodeGroup labels={["script.js"]} lineNumbers={[true]}>
+
+<!-- eslint-skip -->
+
+```javascript
+import { check } from 'k6';
+import { chromium } from 'k6/x/browser';
+
+export default function () {
+  const browser = chromium.launch({ headless: false });
+  const page = browser.newPage();
+
+  page.goto('https://test.k6.io/my_messages.php', { waitUntil: 'networkidle' }).then(() => {
+    // Enter login credentials and login
+    page.locator('input[name="login"]').type('admin');
+    page.locator('input[name="password"]').type('123');
+    
+    // Wait for asynchronous operations to complete
+    return Promise.all([
+      page.waitForNavigation(),
+      page.locator('input[type="submit"]').click(),
+    ]).then(() => {
+      check(page, {
+        'header': page.locator('h2').textContent() == 'Welcome, admin!',
+      });
+    }).finally(() => {
+      page.close();
+      browser.close();
+    });
+  });
```

Comment:
The `finally` better be after the first `then` (`.goto`'s `then`) since `goto` can also throw an exception.

```suggestion
    });
  }).finally(() => {
    page.close();
    browser.close();
  });
```